clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -799,6 +799,19 @@ static bool IsEquivalentExceptionSpec(StructuralEquivalenceContext &Context,
   return true;
 }
 
+// Determine structural equivalence of two instances of
+// HLSLAttributedResourceType::Attributes
+static bool
+IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
+                         const HLSLAttributedResourceType::Attributes &Attrs1,
+                         const HLSLAttributedResourceType::Attributes &Attrs2) {
+  if (Attrs1.ResourceClass != Attrs2.ResourceClass)
```

Comment:
nit: this will have more attributes in the future, so it may be convenient to just use std::tie and grow the list as attributes are added, so something like:
`return std::tie(Attrs1.ResourceClass, Attrs1.IsROV, /*is_cube, is_ms, texture_dimension etc*/) == std::tie(Attrs2.ResourceClass, Attrs2.IsROV);

Reply:
I believe it is better to do the compare without any additional allocations.

Reply:
`std::tie` doesn't incur any allocations. You can expect `std::tie(a, b) == std::tie(c, d)` to generate the same code as `a == b && c == d`. See https://godbolt.org/z/d8x46Gzor

Reply:
Great! I did not know that. I'll use `std::tie` then :).